backend/clubs/views.py

```diff
@@ -1008,6 +1008,28 @@ def get_queryset(self):
         return Advisor.objects.filter(club__code=self.kwargs.get("club_code")).order_by("name")
 
 
+class ValidEventFilter(filters.BaseFilterBackend):
+    """
+    A DRF filter to implement custom filtering logic for the frontend.
+    Filter events that are current and are not created by unapproved clubs.
+    """
+    def filter_queryset(self, request, queryset, view):
+        if view.kwargs.get("club_code") is not None:
+            queryset = queryset.filter(club__code=self.kwargs["club_code"])
+
+        queryset = queryset.filter(club__approved=True)
+
+        now = timezone.now()
+        if view.action in ["list"]:
+            queryset = queryset.filter(end_time__gte=now)
+
+        return (
+            queryset.select_related("club", "creator", )
+                .prefetch_related("club__badges")
+                .order_by("start_time")
+        )
```

Comment:
You should put all of the stuff that the user cannot control in the `get_queryset` method, and put all of the stuff you want the user to be able to alter in the event filter. This way, it makes it easier to use the filter for other purposes.

Reply:
So filter by club_code in get_queryset and filtering by time and approval of clubs in ValidEventFilter?

Reply:
So filter by club_code in get_queryset and filtering by time and approval of clubs in ValidEventFilter?

Reply:
This should actually all be in the `get_queryset` method, actually. You want to put the `start_time` and `end_time` filters in here that filter off of GET parameters.